lab_1/main.py

```diff
@@ -98,22 +207,34 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
-
-
-def load_profile(path_to_file: str) -> dict or None:
-    """
-    Loads a language profile
-    :param path_to_file: a path
-    :return: a dictionary with three keys – name, freq, n_words
-    """
-    pass
-
-
-def save_profile(profile: dict) -> int:
-    """
-    Saves a language profile
-    :param profile: a dictionary
-    :return: 0 if everything is ok, 1 if not
-    """
-    pass
+    try:
+        if not languages:
+            languages = profiles.copy()
+        count = 0
+        score_best = 0
+        profile = profiles[count]
+        score_best_name = profile["name"]
+        score_current = 0
+        check = True
+        for language in languages:
+            if isinstance(language, dict):
+                language = language["name"]
+            if language == profile["name"]:
+                score_current = compare_profiles_advanced(unknown_profile, profile, top_n)["score"]
+                check = False
+            if score_current > score_best:
+                score_best_name = language
+                score_best = score_current
+            elif score_current == score_best:
+                compare = sorted([score_best_name, language])
+                score_best_name = compare[0]
+            count += 1
+            try:
+                profile = profiles[count]
+            except IndexError:
+                pass
+        if check:
+            return None
+        return score_best_name
+    except (UnboundLocalError, TypeError):
+        return None
```

Comment:
this function is hard to read

Reply:
Is there something I should do? I don't really understand what are my actions here.